Rename kubernetes salt module and fix deps

sayf-eddine-scality opened the pull request:
https://github.com/saltstack/salt/pull/51936
Fixes: #923

bert-e commented:
# Hello sayf-eddine-scality,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/metalk8s/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `bugfix/rename-salt-kubernetes` into
`development/2.0`
* [w/2.1/bugfix/rename-salt-kubernetes](https://github.com/scality/metalk8s/compare/development/2.1...w/2.1/bugfix/rename-salt-kubernetes) will be merged into `development/2.1` 
* [w/2.2/bugfix/rename-salt-kubernetes](https://github.com/scality/metalk8s/compare/development/2.2...w/2.2/bugfix/rename-salt-kubernetes) will be merged into `development/2.2` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @cmonfort

* @alexandre-allard-scality

* @gdemonet

* @slaperche-scality

* @ramanduh

* @ballot-scality

* @teddyandrieux

* @tcarmet

* @lucieleonard

* @zempashi

* @sayf-eddine-scality

* @carlito-scality

* @jbertran

* @thomasdanan

* @monpote

* @nootal

* @nicolast

* @alxf

slaperche-scality commented on buildchain/buildchain/salt_tree.py:
```diff
@@ -209,7 +209,7 @@ def task__deploy_salt_tree() -> Iterator[types.TaskDict]:
     Path('salt/_modules/cri.py'),
     Path('salt/_modules/docker_registry.py'),
-    Path('salt/_modules/kubernetes.py'),
+    Path('salt/_modules/kubernetesmod.py'),
```
You should explain in the commit message WHY we must rename the file.

gdemonet replied:
A link to the Saltstack PR/issue could be enough IMHO

slaperche-scality replied:
Yes an no, I really would like to have some info that are readable offline.

slaperche-scality commented on buildchain/buildchain/salt_tree.py:
```diff
@@ -209,7 +209,7 @@ def task__deploy_salt_tree() -> Iterator[types.TaskDict]:
     Path('salt/_modules/cri.py'),
     Path('salt/_modules/docker_registry.py'),
-    Path('salt/_modules/kubernetes.py'),
+    Path('salt/_modules/kubernetesmod.py'),
```
Also, not very fond of `kubernetesmod`, maybe `k8s_utils` or sth like that would be better

slaperche-scality replied:
Ok, I heard the reason for behind this name so let's keep it.

slaperche-scality commented on salt/_modules/metalk8s.py:
```diff
@@ -4,16 +4,19 @@
 '''
 import logging
 
+from salt.exceptions import CommandExecutionError
```
Those changes seem totally unrelated, should be done in a dedicated commit (or dedicated PR).

Or am I wrong and it is related to the renaming?

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @cmonfort

* @alexandre-allard-scality

* @gdemonet

* @slaperche-scality

* @ramanduh

* @ballot-scality

* @teddyandrieux

* @tcarmet

* @lucieleonard

* @zempashi

* @sayf-eddine-scality

* @carlito-scality

* @jbertran

* @thomasdanan

* @monpote

* @nootal

* @nicolast

* @alxf





The following reviewers are expecting changes from the author, or must review again:

* @slaperche-scality

gdemonet reviewed (dismissed):
LGTM

sayf-eddine-scality committed:
Rename kubernetes salt module

The kubernetes salt module has the same name as the python k8s client
thus we run into wrong imports errors, we are renaming the module the
same as it is done in the official salt repo

Fixes: #923

sayf-eddine-scality committed:
Depends on kubernetes lib only when needed

We dont need the k8s module for all the function in the metalk8s salt
module. So we are removing it from the virtual method and adding depends
decorators when needed.

Fixes: #923

bert-e commented:
# History mismatch



Merge commit #27cbd4c50b85369b27275b4a6c32eecde9798981 on the integration branch
`w/2.1/bugfix/rename-salt-kubernetes` is merging a branch which is neither the current
branch `bugfix/rename-salt-kubernetes` nor the development branch
`development/2.1`.

It is likely due to a rebase of the branch `bugfix/rename-salt-kubernetes` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

sayf-eddine-scality commented:
@bert-e reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `bugfix/rename-salt-kubernetes` into
`development/2.0`
* [w/2.1/bugfix/rename-salt-kubernetes](https://github.com/scality/metalk8s/compare/development/2.1...w/2.1/bugfix/rename-salt-kubernetes) will be merged into `development/2.1` 
* [w/2.2/bugfix/rename-salt-kubernetes](https://github.com/scality/metalk8s/compare/development/2.2...w/2.2/bugfix/rename-salt-kubernetes) will be merged into `development/2.2` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @cmonfort

* @alexandre-allard-scality

* @gdemonet

* @slaperche-scality

* @ramanduh

* @ballot-scality

* @teddyandrieux

* @tcarmet

* @lucieleonard

* @zempashi

* @sayf-eddine-scality

* @carlito-scality

* @jbertran

* @thomasdanan

* @monpote

* @nootal

* @nicolast

* @alxf





The following reviewers are expecting changes from the author, or must review again:

* @slaperche-scality

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @cmonfort

* @alexandre-allard-scality

* @gdemonet

* @slaperche-scality

* @ramanduh

* @ballot-scality

* @teddyandrieux

* @tcarmet

* @lucieleonard

* @zempashi

* @sayf-eddine-scality

* @carlito-scality

* @jbertran

* @thomasdanan

* @monpote

* @nootal

* @nicolast

* @alxf

sayf-eddine-scality commented:
@bert-e approve

bert-e commented:
# Build failed



The build for [commit](https://github.com/scality/metalk8s/commit/f953afd5a0346bdfbdb30b6f11b4ce7f35751bed) did not succeed in branch bugfix/rename-salt-kubernetes.




*The following options are set:* **approve**

bert-e commented:
# Build failed



The build for [commit](https://github.com/scality/metalk8s/commit/39ab04067cde680c5f46007e255c8bc6d45b44f6) did not succeed in branch w/2.1/bugfix/rename-salt-kubernetes.




*The following options are set:* **approve**

bert-e commented:
# Build failed



The build for [commit](https://github.com/scality/metalk8s/commit/f953afd5a0346bdfbdb30b6f11b4ce7f35751bed) did not succeed in branch bugfix/rename-salt-kubernetes.




*The following options are set:* **approve**

bert-e commented:
# History mismatch



Merge commit #e7d31d73d837c8d5df0f990dbdeafe8eda1e2a9f on the integration branch
`w/2.1/bugfix/rename-salt-kubernetes` is merging a branch which is neither the current
branch `bugfix/rename-salt-kubernetes` nor the development branch
`development/2.1`.

It is likely due to a rebase of the branch `bugfix/rename-salt-kubernetes` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**





*The following options are set:* **approve**

sayf-eddine-scality commented:
@bert-e reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.






*The following options are set:* **approve**

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `bugfix/rename-salt-kubernetes` into
`development/2.0`
* [w/2.1/bugfix/rename-salt-kubernetes](https://github.com/scality/metalk8s/compare/development/2.1...w/2.1/bugfix/rename-salt-kubernetes) will be merged into `development/2.1` 
* [w/2.2/bugfix/rename-salt-kubernetes](https://github.com/scality/metalk8s/compare/development/2.2...w/2.2/bugfix/rename-salt-kubernetes) will be merged into `development/2.2` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```







*The following options are set:* **approve**

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @cmonfort

* @alexandre-allard-scality

* @gdemonet

* @slaperche-scality

* @ramanduh

* @ballot-scality

* @teddyandrieux

* @tcarmet

* @lucieleonard

* @zempashi

* @sayf-eddine-scality

* @carlito-scality

* @jbertran

* @thomasdanan

* @monpote

* @nootal

* @nicolast

* @alxf










*The following options are set:* **approve**

bert-e commented:
# Build failed



The build for [commit](https://github.com/scality/metalk8s/commit/37ff22e9602e68e6f1300a31611bf68ec411c70c) did not succeed in branch w/2.1/bugfix/rename-salt-kubernetes.




*The following options are set:* **approve**

bert-e commented:
# Build failed



The build for [commit](https://github.com/scality/metalk8s/commit/06e663030654a4354de9c057d8d2b333c6f4fa19) did not succeed in branch w/2.2/bugfix/rename-salt-kubernetes.




*The following options are set:* **approve**

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/2.0`

* :heavy_check_mark: `development/2.1`

* :heavy_check_mark: `development/2.2`



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/2.0`

* :heavy_check_mark: `development/2.1`

* :heavy_check_mark: `development/2.2`



The following branches have **NOT** changed:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`



Please check the status of the associated issue None.

Goodbye sayf-eddine-scality.
